library/src/test/kotlin/build/buf/connect/impl/ProtocolClientTest.kt

```diff
@@ -0,0 +1,48 @@
+package build.buf.connect.impl
+
+import build.buf.connect.Codec
+import build.buf.connect.MethodSpec
+import build.buf.connect.ProtocolClientConfig
+import build.buf.connect.SerializationStrategy
+import okio.Buffer
+import org.junit.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+
+class ProtocolClientTest {
+    private val serializationStrategy: SerializationStrategy = mock { }
+    private val codec: Codec<String> = mock { }
+
+    @Test
+    fun urlConfiguration() {
```

Comment:
I don't see actual validation being done here - am I missing something?

Reply:
The existing behavior fails depending if the user sets the "host" with a trailing `/` or not. This test exercises both those use cases

Reply:
Done